galley/pkg/server/server.go

```diff
@@ -128,17 +128,17 @@ func newServer(a *Args, p patchTable) (*Server, error) {
 		if err != nil {
 			return nil, err
 		}
+		var found []schema.ResourceSpec
+
 		if !a.DisableResourceReadyCheck {
-			if err := p.verifyResourceTypesPresence(k, sourceSchema.All()); err != nil {
-				return nil, err
-			}
+			found, err = p.verifyResourceTypesPresence(k, sourceSchema.All())
```

Comment:
It looks like we should be able to practically reduce verifyResourceTypesPresence & findSupportedResources  into a single method call, with a boolean flag, indicating whether non-optional resources should be waited on to appear, but otherwise simply returns a list of resources that were found.

Then when the output returns with found resources, we can do a check whether if any of the non-optional resources were missing. If so, we can either log or error out, depending on the DisableResourceReadyCheck value.


WDYT?

Reply:
The funciton needs to know which CRDs are required since it does repeated polling. So yes we can merge the two with another argument saying whether to poll for required resources. 

I'm not I understand the second part. If the polling for required resources fails then it is an error and no need to compare found resources. If it succeeds, then we can use the found list in place of  findResourceTypesPresent output.

Reply:
Combined the functions, PTAL.